src/User.php

```diff
@@ -384,11 +385,20 @@ public function roles()
      * </span>
      * ```
      *
+     * @example
+     * Show edit link for posts that a user can edit
+     *
+     * ```twig
+     * {% if user.can('edit_post', post.id) %}
+     *     <a href="{{post.edit_link}}">Edit Post</a>
+     * {% endif %}
+     * ```
+     *
      * @return bool Whether the user has the capability.
      */
-    public function can($capability)
+    public function can($capability, ...$args)
     {
-        return user_can($this->ID, $capability);
+        return user_can($this->ID, $capability, ...$args);
```

Comment:
Since we're now storing core objects (WP_Post, WP_user, etc.) in our models, we should directly pass `$this->wp_object` in the `user_can` first argument. It would avoid WP getting the user (although it's already cached).

See https://github.com/WordPress/wordpress-develop/blob/2bb5679d666474d024352fa53f07344affef7e69/src/wp-includes/capabilities.php#L790-L802

Reply:
Thanks @nlemoine, good catch!

Resolved in #2647.